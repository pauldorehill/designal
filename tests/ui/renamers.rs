use designal::Designal;

#[derive(Designal)]
#[designal(rename)]
struct HumanBean();
#[derive(Designal)]
#[designal(rename = "")]
struct HumanBean1();

#[derive(Designal)]
#[designal(rename = 2)]
struct HumanBean2();

#[derive(Designal)]
#[designal(rename = HumanBean)]
struct HumanBean3();

#[derive(Designal)]
#[designal(rename = "HumanBean1")]
#[designal(rename = "HumanBean2")]
struct HumanBean4();

#[derive(Designal)]
#[designal(rename = "HumanBean1", rename = "HumanBean2")]
struct HumanBean5();

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean6(#[designal(rename = "flavour")] String);

#[derive(Designal)]
#[designal(rename = "HumanBean", add_start = "Snozz")]
struct HumanBean7();

#[derive(Designal)]
#[designal(rename = "HumanBean", add_start = "Snozz", add_end = "Snozz")]
struct HumanBean8();

#[derive(Designal)]
#[designal(
    rename = "HumanBean",
    add_start = "Snozz",
    add_end = "Snozz",
    trim_start = "Snozz"
)]
struct HumanBean9();

#[derive(Designal)]
#[designal(
    rename = "HumanBean",
    add_start = "Snozz",
    add_end = "Snozz",
    trim_start = "Snozz",
    trim_end = "Snozz"
)]
struct HumanBean10();

#[derive(Designal)]
#[designal(trim_start = "Snozz")]
struct HumanBean11();

#[derive(Designal)]
#[designal(trim_end = "Snozz")]
struct HumanBean12();

fn main() {}
