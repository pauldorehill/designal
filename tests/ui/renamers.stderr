error: You need to provide a way to rename the struct like `rename = "NoSignals"
 --> $DIR/renamers.rs:4:12
  |
4 | #[designal(rename)]
  |            ^^^^^^

error: rename can't be empty
 --> $DIR/renamers.rs:7:21
  |
7 | #[designal(rename = "")]
  |                     ^^

error: Only string literals are allowed
  --> $DIR/renamers.rs:11:21
   |
11 | #[designal(rename = 2)]
   |                     ^

error: expected literal
  --> $DIR/renamers.rs:15:21
   |
15 | #[designal(rename = HumanBean)]
   |                     ^^^^^^^^^

error: You should only rename once
  --> $DIR/renamers.rs:20:12
   |
20 | #[designal(rename = "HumanBean2")]
   |            ^^^^^^

error: You should only rename once
  --> $DIR/renamers.rs:24:35
   |
24 | #[designal(rename = "HumanBean1", rename = "HumanBean2")]
   |                                   ^^^^^^

error: You cannot rename a unnamed field
  --> $DIR/renamers.rs:29:30
   |
29 | struct HumanBean6(#[designal(rename = "flavour")] String);
   |                              ^^^^^^

error: You can only do one of rename, add_start, add_end, trim_start, trim_end
  --> $DIR/renamers.rs:32:34
   |
32 | #[designal(rename = "HumanBean", add_start = "Snozz")]
   |                                  ^^^^^^^^^

error: You can only do one of rename, add_start, add_end, trim_start, trim_end
  --> $DIR/renamers.rs:36:55
   |
36 | #[designal(rename = "HumanBean", add_start = "Snozz", add_end = "Snozz")]
   |                                                       ^^^^^^^

error: You can only do one of rename, add_start, add_end, trim_start, trim_end
  --> $DIR/renamers.rs:44:5
   |
44 |     trim_start = "Snozz"
   |     ^^^^^^^^^^

error: You can only do one of rename, add_start, add_end, trim_start, trim_end
  --> $DIR/renamers.rs:54:5
   |
54 |     trim_end = "Snozz"
   |     ^^^^^^^^

error: struct HumanBean11 does not start with Snozz
  --> $DIR/renamers.rs:59:12
   |
59 | #[designal(trim_start = "Snozz")]
   |            ^^^^^^^^^^

error: struct HumanBean12 does not end with Snozz
  --> $DIR/renamers.rs:63:12
   |
63 | #[designal(trim_end = "Snozz")]
   |            ^^^^^^^^

error: `trim_end_all` is only valid at the container level
  --> $DIR/renamers.rs:72:16
   |
72 |     #[designal(trim_end_all = "Human")]
   |                ^^^^^^^^^^^^

error: `trim_start_all` is only valid at the container level
  --> $DIR/renamers.rs:78:16
   |
78 |     #[designal(trim_start_all = "Human")]
   |                ^^^^^^^^^^^^^^
